src/services/DataBase.ts

```diff
@@ -28,7 +33,13 @@ export class DataBase {
 		logger.info(`DataBase Connected to MongoDB`);
 	};
 
-	static getNodeList = ({ filter, limit }: NodeSearchCriteria = { filter: {}, limit: 0 }): Promise<NodeDocument[]> => {
+	static getNodeList = (
+		{ filter, limit, order }: NodeSearchCriteria = { filter: {}, limit: 0, order: NodeListOrder.Random },
+	): Promise<NodeDocument[]> => {
+		if (limit > 0 && order === NodeListOrder.Random) {
+			return Node.aggregate([{ $match: { ...filter } }, { $sample: { size: limit } }]).exec();
```

Comment:
Are we sure that the `$sample` will not return the same document more than once in the result set?

Reply:
Yes =>
https://docs.mongodb.com/manual/reference/operator/aggregation/sample/
> WiredTiger or in-memory storage engine, $sample does not return duplicate documents. WiredTiger is the default storage engine as of MongoDB 3.2.